File: backend/MedTrack.API/MedTrack.API/Models/PatientChronicDisease.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedTrack.API.Models
{
    public class PatientChronicDisease
    {

        [Key, Column(Order = 0)]
        [ForeignKey(nameof(Patient))]
        public int PatientId { get; set; }

        [Key, Column(Order = 1)]
        [ForeignKey(nameof(Disease))]
        public int DiseaseId { get; set; }

        public string? OtherText { get; set; }

        public Patient? Patient { get; set; }
        public ChronicDisease? Disease { get; set; }
    }
}

File: frontend/medtrack-frontend/src/api/patientChronicDisease.js
// import api from './client';

// export function getAllChronicDiseases() {
//   return api.get('/api/ChronicDisease')
//     .then(res => res.data);
// }

// export function getChronicDiseasesByPatient(patientId) {
//   return api.get(`/api/PatientChronicDisease/patient/${patientId}`)
//     .then(res => res.data);
// }

// export function addChronicDisease(payload) {
//   return api.post('/api/PatientChronicDisease', payload)
//     .then(res => res.data);
// }

// export function deleteChronicDisease(payload) {
//   return api.delete('/api/PatientChronicDisease', { data: payload })
//     .then(res => res.data);
// }

// export function updateChronicDisease(diseaseId, payload) {
//   return api.put(`/api/PatientChronicDisease/${diseaseId}`, payload)
//     .then(res => res.data);
// }
import api from './client'

export const listPatientChronicDiseases = (patientId) =>
  api.get(`/api/PatientChronicDisease/patient/${patientId}`).then(r => r.data)

export const addPatientChronicDisease = (payload /* { patientId, diseaseId, otherText? } */) =>
  api.post('/api/PatientChronicDisease', payload).then(r => r.data)

export const removePatientChronicDisease = ({ patientId, diseaseId }) =>
  api.delete('/api/PatientChronicDisease', {
    params: { patientId, diseaseId }
  }).then(r => r.data)

